server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -7,138 +7,294 @@ import { assert } from "console";
 import * as core from "@fluidframework/server-services-core";
 import { AggregationCursor, Collection, MongoClient, MongoClientOptions } from "mongodb";
 import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import { runWithRetry } from "@fluidframework/server-services-core";
+import { MongoErrorRetryAnalyzer } from "./mongoExceptionRetryRules/MongoErrorRetryAnalyzer";
+
 
 const MaxFetchSize = 2000;
 
 export class MongoCollection<T> implements core.ICollection<T> {
-    constructor(private readonly collection: Collection<T>) {
+    constructor(
+        private readonly collection: Collection<T>,
+        private readonly clientSideRetry?: boolean,
+    ) {
+        assert(collection, "collection must be defined");
     }
 
     public aggregate(pipeline: any, options?: any): AggregationCursor<T> {
-        return this.collection.aggregate(pipeline, options);
+        return runWithRetry(
```

Comment:
Nit: Might be useful to create a version of runWithRetry for mongo with default maxReties, retryAfterMs, telemetryProperties, shouldIgnoreError and shouldRetry so that the callers become simpler and the cases where we are are deviating from the defaults become more explicit.